subsquid/src/dbHelper.ts

```diff
@@ -248,6 +249,60 @@ export async function storeHistoricalLockedValue(
   await ctx.store.save(historicalLockedValueSource);
 }
 
+/**
+ * Stores a new HistoricalVolume for the specified quote asset
+ * @param ctx
+ * @param quoteAssetId
+ * @param amount
+ */
+export async function storeHistoricalVolume(
+  ctx: EventHandlerContext<Store>,
+  quoteAssetId: string,
+  amount: bigint
+): Promise<void> {
+  const wsProvider = new WsProvider(chain());
+  const api = await ApiPromise.create({ provider: wsProvider });
+  let assetPrice = 0n;
+
+  try {
+    const oraclePrice = await api.query.oracle.prices(quoteAssetId);
+    console.log(oraclePrice);
+    if (!oraclePrice?.price) {
+      // TODO: handle missing oracle price
+      // NOTE: should we look at the latest known price for this asset?
+      return;
+    }
+    assetPrice = BigInt(oraclePrice.price.toString());
```

Comment:
@andresmechali we need to add a decimal conversion here otherwise therell be precision issue

Reply:
yes, I will add it once https://github.com/ComposableFi/composable/pull/2043 is merged